backend.c

```diff
@@ -1428,9 +1441,10 @@ static uint64_t do_dry_run(struct thread_data *td)
 			td->ts.total_io_u[io_u->ddir]++;
 		}
 
-		if (td_write(td) && io_u->ddir == DDIR_WRITE &&
-		    td->o.do_verify &&
-		    td->o.verify != VERIFY_NONE &&
+		// TODO Setting verify_backlog could result in verify reads from get_io_u
+		// TODO that would not be performed when we fake completion and thus not all writes
+		// TODO would be verified in a dry run.
+		if (verifiable_io(td, io_u) &&
```

Comment:
It would be useful to bail out with an explanation at startup time if your constraints aren't met e.g. in `fixup_options()` of init.c .